frontend/test/metabase/scenarios/dashboard/reproductions/20637-add-series-to-dashcard.cy.spec.js

```diff
@@ -0,0 +1,95 @@
+import { restore } from "__support__/e2e/cypress";
+
+import { SAMPLE_DATABASE } from "__support__/e2e/cypress_sample_database";
+
+const { PRODUCTS, PRODUCTS_ID } = SAMPLE_DATABASE;
+
+describe("adding an additional series to a dashcard (metabase#20637)", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("should use the correct query endpoints (metabase#20637)", () => {
+    createQuestionsAndDashboard();
+    cy.wait("@dashcardQuery");
+
+    // edit the dashboard and open the add series modal
+    cy.icon("pencil").click();
+    // the button is made clickable by css using :hover so we need to force it
+    cy.findByTestId("add-series-button").click({ force: true });
+
+    cy.findByText("20637 Question 2").click();
+    // make sure the card query endpoint was used
+    cy.wait("@additionalSeriesCardQuery");
+
+    cy.get(".AddSeriesModal").within(() => {
+      cy.findByText("Done").click();
+    });
+    cy.findByText("Save").click();
+
+    // refresh the page and make sure the dashcard query endpoint was used
+    cy.reload();
```

Comment:
It is slightly concerning that the reload happens faster than the `GET /api/dashboard/2` so it cancels that request.
![image](https://user-images.githubusercontent.com/31325167/155536751-58edb17f-7eb5-4753-92f0-fd8188b78810.png)

Just to be on the safe side, maybe it's worth waiting for all requests before reloading. Something like this:
```js
function saveDashboard() {
  cy.intercept("PUT", "/api/dashboard/*").as("updateDashboard");
  cy.intercept("PUT", "/api/dashboard/*/cards").as("updateDashCards");
  cy.intercept("GET", "/api/dashboard/*").as("loadDashboard");

  cy.findByText("Save").click();

  cy.wait(["@updateDashboard", "@updateDashCards", "@loadDashboard"]);
}
```

If I do that locally, there are no cancelled requests any more:
![image](https://user-images.githubusercontent.com/31325167/155537529-139f541d-a5c3-4ee4-8970-c0b9fcaf58f7.png)

Reply:
yeah, seems like a good idea to add